lib/sensors/stripe_sensor.hpp

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include <cstdint>
+#include <optional>
+
+#include "io/gpio.hpp"
+#include <core/logger.hpp>
+#include <io/hardware_gpio.hpp>
+
+namespace hyped::sensors {
+class Keyence {
+ public:
+  Keyence(core::ILogger &log, const std::uint8_t newPin);
+  ~Keyence();
+
+  int getStripeCount();
+  void updateStripes();
+
+ private:
+  std::uint8_t pin;
+  uint16_t stripe_count_;
+  std::optional<std::shared_ptr<IGpioReader>> keyence;

```

Comment:
Same here, `keyence_`.

Reply:
actually, this should be called `gpio_reader_` and not `keyence_`